### ISSUE TITLE
FinalizedBy ordering rule is not enforced in Gradle 4.8-rc-1

### ISSUE THREAD

**Issue by mark-vieira** [gradle/gradle#5415]
### Expected Behavior

When multiple tasks are `finalizedBy` the same task, the finalizer should not run until all tasks that it "finalizes" have completed (or failed). In conjunction with a multi-project build and parallel execution this should also apply to cross-project tasks. That is, if `a:foo` and `b:foo` are both `finalizedBy` `a:finalizer` then `a:finalizer` should not run before `a:foo` and `b:foo` have completed, even if they run concurrently.

### Current Behavior

This looks to be a regression in Gradle 4.8 as the behavior I describe above is the case in Gradle 4.7. In 4.8 the finalizer task is run concurrently with `b:foo` (in the example above).

### Steps to Reproduce (for bugs)

Here's a sample project that reproduces the bug:

```
configure(project(':a')) {
    task finalizer {
        doLast {
            sleep 100
        }
    }
    
    task foo {
        finalizedBy finalizer
        doLast {
            sleep 500
        }
    }
}

configure(project(':b')) {
    task foo {
        finalizedBy ':a:finalizer'
        doLast {
            sleep 1000
        }
    }
}
```

### Your Environment

Here are build scans of the above example build:
Gradle 4.7: https://scans.gradle.com/s/xdqowj2igjbcg
Gradle 4.8-rc-1: https://scans.gradle.com/s/kvbhpwitvaymo

Notice the difference in the timeline.

**Comment by wolfs** [comment]
@mark-vieira Thank you for reporting - I'll look into this right now.

**Comment by wolfs** [comment]
PR for fixing it: https://github.com/gradle/gradle/pull/5417

**Comment by lptr** [comment]
What broke this?

**Comment by wolfs** [comment]
This behaviour was broken by https://github.com/gradle/gradle/pull/5153.

**Comment by wolfs** [comment]
Fixed via https://github.com/gradle/gradle/pull/5417. The fix will be in 4.8 RC2.
